src/server/bg_services/log_replication_scanner.js

```diff
@@ -156,6 +155,12 @@ class LogReplicationScanner {
         }
     }
 
+    async process_copy_candidate(src_bucket, dst_bucket, candidate) {
+        if (candidate.src_object_info && candidate.dst_object_info) {
```

Comment:
if both exist - then if dest is newer, we should skip replication. if the source is newer we should compare the etags and if they are the same we should skip replication. otherwise, we should replicate

Reply:
But won't destination be more often than not newer than the source because of the log lag?

For example:

1. [12:00 AM] PUT /key1
2. [02:00 AM] PUT /key1
3. [02:05 AM] Log entry corresponding to 12:00 AM entry shows up.
4. [02:05 AM] Processed the entry, created a replica.
5. [03:00 AM] Log entry corresponding to 02:00 AM entry shows up => Now here the the creation time of source object < creation time of destination object but that does not imply that destination object is necessarily up to date.

Reply:
@utkarsh-pro regarding number 5, the key1 in dest is the same data as the one put in 2 `[02:00 AM] PUT /key1`, since we always copy the latest one. also it will match the etag.
so we can keep comparing the create time and it should be ok. isn't it?

Reply:
I think this condition would work here:

```suggestion
        if (src_object_info &&  (!dst_object_info ||  (src_object_info.LastModified > dst_object_info.LastModified ||
            src_object_info.ETag !== dst_object_info.ETag))) {
            await this.copy_object(src_bucket.name, dst_bucket.name, candidate.key);
         }
```

Pls correct me if I am wrong, we are updating the dst object here in both the cases even if src_LastModified > dst_LastModified or Etag not matches. This condition would work fine if the logs are sorted/sequential.